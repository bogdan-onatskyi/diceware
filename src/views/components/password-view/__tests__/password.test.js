import React from 'react';

import Password from '../password';
import passwordObject from '../../../../models/password-object';

describe("Password component", () => {
    const setup = ({
                       password = passwordObject.password,
                       isVariantsOpened = passwordObject.isVariantsOpened,
                       toggleVariantsOpened = passwordObject.toggleVariantsOpened
                   }) => {

        const wrapper = shallow(<Password password={password}
                                          isVariantsOpened={isVariantsOpened}
                                          toggleVariantsOpened={toggleVariantsOpened}/>
        );
        return {wrapper};
    };

    it("should match saved snapshot (passwordObject.isVariantsOpened === false)", () => {
        const {wrapper} = setup({
            password: 'word1 word2 word3',
            isVariantsOpened: false
        });
        expect(wrapper).toMatchSnapshot();
    });

    it("should match saved snapshot (passwordObject.isVariantsOpened === true)", () => {
        const {wrapper} = setup({
            password: 'word1 word2 word3',
            isVariantsOpened: true
        });
        expect(wrapper).toMatchSnapshot();
    });

    it("should toggle passwordObject.isVariantsOpened", () => {
        passwordObject.isVariantsOpened = false;
        const {wrapper} = setup({});

        wrapper.find('.password__before-text').simulate('click');
        expect(passwordObject.isVariantsOpened).toBe(true);

        wrapper.find('.password__before-text').simulate('click');
        expect(passwordObject.isVariantsOpened).toBe(false);
    });

    it("should have 3 props", () => {
        const propsStr = 'password isVariantsOpened toggleVariantsOpened';
        const props = [];
        for (let prop in Password.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});