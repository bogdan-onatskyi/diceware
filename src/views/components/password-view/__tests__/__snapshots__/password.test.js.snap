// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password component should match saved snapshot (passwordObject.isVariantsOpened === false) 1`] = `
<div
  className="password"
>
  <span
    className="password__before-text icon-right-open"
    data-tip="Варианты пароля с разделителями"
    onClick={[Function]}
  />
  <div
    className="password__text"
    data-tip="Нажмите на иконку справа для копирования пароля в буфер обмена"
    id="pass"
  >
    word1 word2 word3
  </div>
  <span
    className="clipboard password__after-text icon-copy"
    data-clipboard-target="#pass"
    data-delay-hide={500}
    data-event="click"
    data-iscapture={true}
    data-tip="Пароль скопирован в буфер обмена"
  />
  <ToolTip />
</div>
`;

exports[`Password component should match saved snapshot (passwordObject.isVariantsOpened === true) 1`] = `
<div
  className="password"
>
  <span
    className="password__before-text icon-down-open"
    data-tip="Варианты пароля с разделителями"
    onClick={[Function]}
  />
  <div
    className="password__text"
    data-tip="Нажмите на иконку справа для копирования пароля в буфер обмена"
    id="pass"
  >
    word1 word2 word3
  </div>
  <span
    className="clipboard password__after-text icon-copy"
    data-clipboard-target="#pass"
    data-delay-hide={500}
    data-event="click"
    data-iscapture={true}
    data-tip="Пароль скопирован в буфер обмена"
  />
  <ToolTip />
</div>
`;
